### PR TITLE
APP-329: [C/C++] Distinguish between CMake variables and compile definitions in cmakegen

### PR DESCRIPTION
**Pull request by RostakaGmfun**
/cc @Slash32 @vadimol

### CODE REVIEW

File: credentialsdemo/source/c/CMakeLists.txt [hunk 1]
```diff
@@ -23,17 +23,15 @@ set(WITH_EXTENSION_NOTIFICATIONS OFF)
 set(WITH_EXTENSION_LOGGING OFF)
 set(WITH_EXTENSION_EVENTS OFF)
 
+
+if (NOT DEFINED KAA_MAX_LOG_LEVEL)
+    set(KAA_MAX_LOG_LEVEL "3")
+endif (NOT DEFINED KAA_MAX_LOG_LEVEL)
 if (NOT DEFINED KAA_RUNTIME_KEY_GENERATION)
```
Comment (rasendubi) [inline credentialsdemo/source/c/CMakeLists.txt:30]: It's default `OFF` in the C SDK, so you may drop this.

File: configurationdemo/source/c/CMakeLists.txt [hunk 2]
```diff
@@ -26,23 +26,19 @@ if (NOT DEFINED WIFI_SSID)
...
 add_definitions(-DWIFI_PWD=${WIFI_PWD})
 
-if (NOT DEFINED KAA_TARGET)
-    set(KAA_TARGET "posix")
-endif (NOT DEFINED KAA_TARGET)
-add_definitions(-DKAA_TARGET=${KAA_TARGET})
-
-
-
+if (NOT DEFINED KAA_PLATFORM)
```
Comment (rasendubi) [inline configurationdemo/source/c/CMakeLists.txt:34]: It's default in the C SDK

File: configurationdemo/source/c/CMakeLists.txt [hunk 3]
```diff
@@ -26,23 +26,19 @@ if (NOT DEFINED WIFI_SSID)
...
+    set(KAA_PLATFORM "posix")
+endif (NOT DEFINED KAA_PLATFORM)
 if (NOT DEFINED KAA_MAX_LOG_LEVEL)
-set(KAA_MAX_LOG_LEVEL 3)
+    set(KAA_MAX_LOG_LEVEL "3")
```
Comment (rasendubi) [inline configurationdemo/source/c/CMakeLists.txt:38]: Why string?

**Review by rasendubi — COMMENTED** [review]
Why don't we use existing template engine solutions instead of rolling our own? NIH?

**Comment by rasendubi** [comment]
https://www.simple-is-better.org/template/ has good list of template engines

**Review by Slash32 — CHANGES_REQUESTED** [review]
The questions must be addressed before merge.

**Comment by Slash32** [comment]
The PR is considered incomplete and needs re-do. The further changes will be done separately.

**Pull request closed without merging**